src/dataflow-types/src/client.rs

```diff
@@ -658,6 +667,120 @@ pub mod partitioned {
 
     use timely::progress::frontier::MutableAntichain;
 
+    /// Buffer for partial results for a `TAIL`.
+    /// This exists so we can consolidate rows and
+    /// sort them by timestamp before passing them to the consumer
+    /// (currently, coord.rs).
+    struct PendingTail {
+        /// `frontiers[i]` is an antichain representing the times at which we may
+        /// still receieve results from worker `i`.
+        frontiers: HashMap<usize, Antichain<Timestamp>>,
+        /// The results (possibly unsorted) which have not yet been delivered.
+        buffer: Vec<(Row, Timestamp, Diff)>,
+        /// The number of unique shard IDs expected.
+        parts: usize,
+        /// The last progress message that has been reported to the consumer.
+        reported_frontier: Antichain<Timestamp>,
+    }
+
+    impl PendingTail {
+        // See [`differential_dataflow::consolidation::consolidate_updates_slice`].
+        // This function works similarly, except that we sort by timestamp first, then data.
+        //
+        // I implemented it without `unsafe`; we can revisit that if proves to be
+        // a hot path.
+        /// Return all the updates with time less than `upper` (or all times if that is `None`),
+        /// in sorted and consolidated representation.
+        pub fn consolidate_up_to(
```

Comment:
This is in other locations called `seal`, and maybe it makes sense to use the same name here?